session.go

```diff
@@ -3827,19 +3827,20 @@ func (iter *Iter) Timeout() bool {
 func (iter *Iter) Next(result interface{}) bool {
 	iter.m.Lock()
 	iter.timedout = false
-	timeout := time.Time{}
+	deadline := time.Time{}
 
 	// check should we expect more data.
 	for iter.err == nil && iter.docData.Len() == 0 && (iter.docsToReceive > 0 || iter.op.cursorId != 0) {
-		// we should expect more data.
+		// either there is a getMore running or we will schedule one, set
+		// the deadline regardless.
+		if iter.timeout >= 0 && deadline.IsZero() {
```

Comment:
Super minor nitpick, but assuming iter.timeout is usually true and deadline.IsZero is usually false (after the first time through the loop), then reversing the predicate order will shortcut faster.

Reply:
But then it will shortcut slower for the more general case where there is no timeout at all.

Reply:
I'd assume a timeout is more common, but it doesn't really matter in the real world.

Reply:
I believe timeouts are only set for tailable cursors that's why I think it would be the minority.